File: src/main/java/cookiedragon/luchadora/event/entity/SendMessageEvent.java
package cookiedragon.luchadora.event.entity;

import cookiedragon.luchadora.event.CancellableEvent;

/**
 * @author cookiedragon234 08/Dec/2019
 */
public class SendMessageEvent extends CancellableEvent
{
	private final String message;
	
	public SendMessageEvent(String message)
	{
		this.message = message;
	}
	
	public String getMessage()
	{
		return this.message;
	}
	
	public static class Pre extends SendMessageEvent
	{
		public Pre(String message)
		{
			super(message);
		}
	}
	
	public static class Post extends SendMessageEvent
	{
		public Post(String message)
		{
			super(message);
		}
	}
}

File: src/main/kotlin/cookiedragon/luchadora/event/entity/SendMessageEvent.kt
package cookiedragon.luchadora.event.entity

import cookiedragon.luchadora.event.CancellableEvent

/**
 * @author cookiedragon234 17/Feb/2020
 */
open class SendMessageEvent(open val message: String) : CancellableEvent() {
	data class Pre(override val message: String) : SendMessageEvent(message)
	data class Post(override val message: String) : SendMessageEvent(message)
}
